pegasus/src/homepage.rb

```diff
@@ -31,6 +31,12 @@ def self.get_announcement_for_page(page, request)
       # If the banner has an array of languages, then the current language must be one of them.
       next if banner["languages"] && !banner["languages"].include?(request.language)
 
+      # If the banner is shown internationally don't show in the US.
+      # If the banner is not shown internationally, show in the US.
+      location = Geocoder.search(request.ip)&.first
+      in_us = location&.country_code.to_s.casecmp?('us')
+      next unless banner["showInternationally"] ^ in_us
```

Comment:
questions:
1. how often is request.ip set? is it as reliable as user.current_sign_in_ip?
2. does this codepath work (no errors) when request.ip is nil?
3. in the case where request.ip is nil, are we ok with showing only outside us?
4. I'm not sure about taking away the option to display banners globally. isn't this something we'll want to do for HOC?

Reply:
Good questions.
1-2) We can't use current_sign_in_ip because the user may not be signed in and we're on pegasus. But I don't know how often request.ip is set.
I went to this thread https://codedotorg.slack.com/archives/C0T0PNTM3/p1600671794018400 and it looks like the most established pattern is to be using `request.country` instead of `request.location.country_code`. @kelbyhawn thoughts about trying the following instead?
```
# remove location line
in_us = request.country.to_s.casecmp?('us')
next unless banner["showInternationally"] ^ in_us
```
3) @dmcavoy  do you know preferred behavior when we don't know the user's country?
4) There are three options with the `showInternationally` flag:
- `false` –– only show in the US
- `true` –– only show outside of the US
- no flag –– show everywhere

Reply:
> 4. There are three options with the `showInternationally` flag:
> 
> * `false` –– only show in the US
> * `true` –– only show outside of the US
> * no flag –– show everywhere

oh interesting! how is "show everywhere" implemented? the behavior I'm seeing makes me think that "no flag" will behave like `false`:
```
irb(main):001:0> nil ^ false
=> false
irb(main):002:0> nil ^ true
=> true
```

Reply:
If we don't know a users country I think it would be good to default to the US banner.

Reply:
Oh yikes, Dave! Good call –– we tested this manually, but I think didn't cover all the cases.

@kelbyhawn I think either (a) we pair again to get this working, or (b) I pull down the branch and make the commit myself. Do you have a preference?

Reply:
@megcrenshaw , given the need for thorough validation on our side, I'd encourage you to pull down the branch and make any changes needed, rather than going through Kelby for each update.

Reply:
I might go ahead and add unit tests for this part too –– it will help me believe it's working

Reply:
That works for me @megcrenshaw, and thanks @davidsbailey for reviewing this!